azerothcore/azerothcore-wotlk
# Core/CreatureAI - fix enemy stealth alert ignoring los

## Changes Proposed:

Npc's should not get triggered and play alert sound, when player is in stealth but not within line of sight.

## Issues Addressed:
- 

## SOURCE:
I did the same change for VMangos:
https://github.com/vmangos/core/pull/1052

## Tests Performed:
builds, checked ingame


## How to Test the Changes:

1) create level 15 rogue and learn stealth
2) .tele fenris
3) enter castle building, clear enemies in ground level
4) cast stealth (with gm off) and walk around in ground level
5) enemies from floor above should not get triggered and play stealth detect sound


## Known Issues and TODO List:


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
